app/src/main/java/com/snaggly/ksw_toolkit/core/service/mcu/parser/LightEventSwitch.kt

```diff
@@ -1,26 +1,35 @@
 package com.snaggly.ksw_toolkit.core.service.mcu.parser
 
 import android.app.UiModeManager
+import android.content.Context
+import android.provider.Settings
 import com.snaggly.ksw_toolkit.core.service.mcu.McuLogic
 import com.snaggly.ksw_toolkit.core.service.mcu.parser.interfaces.ILightEvent
 import com.snaggly.ksw_toolkit.util.list.eventtype.EventManagerTypes
+import com.wits.pms.utils.BrightnessUtils
 import projekt.auto.mcu.ksw.serial.collection.McuCommands
 import kotlin.experimental.and
+import kotlin.math.roundToInt
 
 object LightEventSwitch : ILightEvent {
 
-    var uiModeManager : UiModeManager? = null
+    var uiModeManager: UiModeManager? = null
 
-    override fun getCarDataEvent(data: ByteArray): EventManagerTypes {
+    override fun getCarDataEvent(data: ByteArray, context: Context): EventManagerTypes {
```

Comment:
See my comment in LightEvent.kt. Same applies here, you don't have to pass the context all around the code.

Reply:
I was able to remove the context from getCarDataEvent()